python/ray/serve/_private/autoscaling_state.py

```diff
@@ -265,18 +270,203 @@ def get_decision_num_replicas(
 
         return self.apply_bounds(decision_num_replicas)
 
-    def get_total_num_requests(self) -> float:
-        """Get average total number of requests aggregated over the past
-        `look_back_period_s` number of seconds.
+    def _collect_replica_running_requests(self) -> List[Dict[str, List]]:
+        """Collect running requests metrics from replicas for aggregation."""
+        metrics_timeseries_dicts = []
+
+        for replica_id in self._running_replicas:
+            if replica_id in self._replica_requests:
+                metrics_timeseries_dicts.append(
+                    {
+                        metric_name: self._replica_requests[replica_id].metrics[
```

Comment:
This will do a lookup into `self._replica_request` every iteration of the comprehension, which is unnecessary since it's the same `replica_id` each time - I'd suggest pulling out 
```python
replica_requests = self._replica_requests[replica_id]
```
and doing a try/catch on the potential key error (which I assume can occur if `self._running_replicas` and `self._replica_requests` are not perfectly in sync?), then re-using that